ICAFIX/scripts/functionhighpassandvariancenormalize.m

```diff
@@ -129,26 +137,42 @@ function functionhighpassandvariancenormalize(TR,hp,fmri,WBC,varargin)
 
 elseif hp>0  % "fslmaths -bptf" based filtering
     if dovol > 0
+        % Save and filter confounds, as a NIFTI, as expected by FIX
         save_avw(reshape(confounds',size(confounds,2),1,1,size(confounds,1)),[fmri hpstring '.ica/mc/prefiltered_func_data_mcf_conf'],'f',[1 1 1 TR]);
         call_fsl(sprintf(['fslmaths ' fmri hpstring '.ica/mc/prefiltered_func_data_mcf_conf -bptf %f -1 ' fmri hpstring '.ica/mc/prefiltered_func_data_mcf_conf_hp'],0.5*hp/TR));
 
-        save_avw(reshape(cts,ctsX,ctsY,ctsZ,ctsT),[fmri hpstring '.nii.gz'],'f',[1 1 1 TR]);
-        call_fsl(['fslmaths ' fmri hpstring '.nii.gz -bptf ' num2str(0.5*hp/TR) ' -1 ' fmri hpstring '.nii.gz']);
-        cts=single(read_avw([fmri hpstring '.nii.gz']));
-        cts=reshape(cts,ctsX*ctsY*ctsZ,ctsT);
-        call_fsl(['fslcpgeom ' fmri '.nii.gz ' fmri hpstring '.nii.gz -d']);
+        % bptf filtering; no masking here, so this is probably memory inefficient
+        call_fsl(['fslmaths ' fmri '.nii.gz -bptf ' num2str(0.5*hp/TR) ' -1 ' fmri hpstring '.nii.gz']);
+        call_fsl(['fslcpgeom ' fmri '.nii.gz ' fmri hpstring '.nii.gz -d']); 
+        
+        % Load in result and reduce to just the non-zero voxels (for memory efficiency going forward)
+        ctsfull=single(read_avw([fmri hpstring '.nii.gz']));
+        ctsX=size(ctsfull,1); ctsY=size(ctsfull,2); ctsZ=size(ctsfull,3); ctsT=size(ctsfull,4); 
+        ctsfull=reshape(ctsfull,ctsX*ctsY*ctsZ,ctsT);
+        ctsmask=std(ctsfull,[],2)>0; 
+        cts=ctsfull(ctsmask,:); 
+        clear ctsfull;
     end
     
     BOdimX=size(BO.cdata,1);  BOdimZnew=ceil(BOdimX/100);  BOdimT=size(BO.cdata,2);
     save_avw(reshape([BO.cdata ; zeros(100*BOdimZnew-BOdimX,BOdimT)],10,10,BOdimZnew,BOdimT),'Atlas','f',[1 1 1 TR]);
     call_fsl(sprintf('fslmaths Atlas -bptf %f -1 Atlas',0.5*hp/TR));
     grot=reshape(single(read_avw('Atlas')),100*BOdimZnew,BOdimT);  BO.cdata=grot(1:BOdimX,:);  clear grot;
-    call_fsl('rm Atlas.nii.gz');
     ciftisave(BO,[fmri '_Atlas' regstring hpstring '.dtseries.nii'],WBC);
+    call_fsl('rm Atlas.nii.gz');
```

Comment:
I suggest `delete('Atlas.nii.gz');` instead, no need to resort to the shell here.

Reply:
Other things in the pipelines matlab use delete(), so I have changed this one also.  We could add a special delete function that bypasses matlab's recycle preference behavior, similar to this:

https://github.com/Washington-University/cifti-matlab/blob/master/private/mydelete.m

Reply:
I don't think we need to be too worried about by-passing the recycle option.  As a practical matter, since we are deleting a file with a fixed (unchanging), the recycle folder is not going to grow without bound.  (Unless the matlab recycle option actually renames the recycled version in the event of a name conflict with an existing file in the recycle folder).

Reply:
The current cifti* functions and save/read_avw in the pipelines also use delete(), so yes, this would be a minor offender even if there is an issue.